eureka2-client/src/main/java/com/netflix/eureka2/client/resolver/EurekaServerResolver.java

```diff
@@ -119,20 +131,34 @@ public EurekaServerResolver(
      * exactly one item with list of retrieved servers, or an empty list, or an error.
      */
     private Observable<Set<Server>> fetchFreshServerSet() {
-        return snapshotInterestChannel
-                .forSnapshot(readServerInterest)
-                .reduce(new HashSet<Server>(), new Func2<Set<Server>, InstanceInfo, Set<Server>>() {
-                    @Override
-                    public Set<Server> call(Set<Server> servers, InstanceInfo newInstanceInfo) {
-                        InetSocketAddress socketAddress = serviceSelector.returnServiceAddress(newInstanceInfo);
-                        Server newServer = new Server(socketAddress.getHostString(), socketAddress.getPort());
-                        servers.add(newServer);
-                        return servers;
-                    }
-                })
-                .retryWhen(new RetryStrategyFunc(500, 2, true))
-                .concatWith(Observable.just(Collections.<Server>emptySet()))
-                .take(1);
+        final EurekaClient eurekaClient = eurekaClientBuilder.build();
+        try {
```

Comment:
Why do we need to try catch here? It's all rx so the doOnTerminate (we can be even safer with a finallyDo) should handle the client closure.